src/sentry/profiles/task.py

```diff
@@ -974,18 +976,11 @@ def _calculate_duration_for_sample_format_v1(profile: Profile) -> int:
 
 
 def _calculate_duration_for_sample_format_v2(profile: Profile) -> int:
-    samples_by_thread = defaultdict(list)
-    for sample in profile["profile"]["samples"]:
-        samples_by_thread[sample["thread_id"]].append(sample["timestamp"])
-    max_duration_ms = 0
-    for samples in samples_by_thread.values():
-        if len(samples) < 2:
-            continue
-        first, last = samples[0], samples[len(samples) - 1]
-        duration_ms = (last - first) * 1e3
-        if duration_ms > max_duration_ms:
-            max_duration_ms = duration_ms
-    return int(max_duration_ms)
+    samples = profile["profile"]["samples"]
+    if len(samples) < 2:
+        return 0
+    first, last = samples[0], samples[-1]
```

Comment:
I feel like we have some similar code in vroom that sorts the samples before taking the first/last because the samples arent guaranteed to come in order but if I'm wrong, just ignore this comment.

Reply:
We can sort to be sure, I don't think that's needed though as samples are an ordered list and time-based and in case of a timestamp bug on the client, we wouldn't necessarily make the situation better.

Reply:
It's not necessarily a "bug".

The Rust profiler, for example, used the `std::time::SystemTime`, which is not monotonic.

> Distinct from the [Instant](https://doc.rust-lang.org/std/time/struct.Instant.html) type, this time measurement is not monotonic. This means that you can save a file to the file system, then save another file to the file system, and the second file has a SystemTime measurement earlier than the first. In other words, an operation that happens after another operation in real time may have an earlier SystemTime!

And if I remember correctly Python samples could also be unsorted although I don't remember what was the reason in that case.